metl/model_encoder.py

```diff
@@ -0,0 +1,114 @@
+import torch
+from typing import Literal
+from biopandas.pdb import PandasPdb
+import metl.relative_attention as ra
+from Bio.SeqUtils import seq1
+import os
+
+class ModelEncoder(torch.nn.Module):
+    def __init__(self, model, encoder, strict=True, indexing:Literal[0,1] = 0) -> None:
+        super(ModelEncoder, self).__init__()
+
+        if indexing != 0 and indexing != 1:
+            raise Exception("Indexing must be equal to 0 or to 1.")
+
+        self.model = model
+        self.encoder = encoder
+
+        self.indexing = indexing
+        self.strict = strict
+
+        self.needs_pdb = self.check_if_pdb_needed(model)
+
+    def check_if_pdb_needed(self, model):
+        sequential = next(model.children())
+        
+        for layer in sequential:
```

Comment:
To make sure I'm reading this correctly, it will iterate through the model layers checking for the relative transformer encoder because that is the indication we have a 3D model.

Reply:
Yea, this is the standard presented in the forward pass of SequentialWithArgs so I used it to know if a PDB file is required. Which it should only be in this case because you need a PDB to do relative encodings.